src/mlpack/methods/ne/genome.hpp

```diff
@@ -0,0 +1,329 @@
+  }
+
+  // Destructor.
+  ~Genome() {}
+
+  // Operator =.
+  Genome& operator =(const Genome& genome) {
+    if (this != &genome) {
+      aId = genome.aId;
+      aNeuronGenes = genome.aNeuronGenes;
+      aLinkGenes = genome.aLinkGenes;
+      aNumInput = genome.aNumInput;
+      aNumOutput = genome.aNumOutput;
+      aFitness = genome.aFitness;
+      aAdjustedFitness = genome.aAdjustedFitness;
+    }
+
+    return *this;
+  }
+
+  // Get genome id.
+  ssize_t Id() const { return aId; }
+
+  // Set genome id.
+  void Id(ssize_t id) { aId = id; }
+
+  // Get input length.
+  ssize_t NumInput() {
+    ssize_t numInput = 0;
+    for (ssize_t i=0; i<aNeuronGenes.size(); ++i) {
+      if (aNeuronGenes[i].Type() == INPUT || aNeuronGenes[i].Type() == BIAS)
+        ++numInput;
+    }
+    return numInput;
+  }
+
+  // Set input length.
+  void NumInput(ssize_t numInput) { aNumInput = numInput; }
+
+  // Get output length.
+  ssize_t NumOutput() {
+    ssize_t numOutput = 0;
+    for (ssize_t i=0; i<aNeuronGenes.size(); ++i) {  
+      if (aNeuronGenes[i].Type() == OUTPUT)
+        ++numOutput;
+    }
+    return numOutput;
+  }
+
+  // Set output length.
+  void NumOutput(ssize_t numOutput) { aNumOutput = numOutput; }
+
+  // Set fitness.
+  void Fitness(double fitness) { aFitness = fitness; }
+
+  // Get fitness.
+  double Fitness() const { return aFitness; }
+
+  // Set adjusted fitness.
+  void AdjustedFitness(double adjustedFitness) { aAdjustedFitness = adjustedFitness; }
+
+  // Get adjusted fitness.
+  double AdjustedFitness() const { return aAdjustedFitness; }
+
+  // Get neuron number.
+  ssize_t NumNeuron() const {
+    return aNeuronGenes.size();
+  }
+  
+  // Get link number.
+  ssize_t NumLink() const {
+    return aLinkGenes.size();
+  }
+
+  // Whether specified neuron id exist in this genome.
+  bool HasNeuronId(ssize_t id) const {
+    for (ssize_t i=0; i<NumNeuron(); ++i) {
+      if (aNeuronGenes[i].Id() == id) {
+        return true;
+      }
+    }
+    
+    return false;
+  }
+
+  // Get neuron by id.
+  NeuronGene GetNeuronById(ssize_t id) {
+    assert(HasNeuronId(id));
+
+    for (ssize_t i=0; i<NumNeuron(); ++i) {
+      if (aNeuronGenes[i].Id() == id) {
+        return aNeuronGenes[i];
+      }
+    }
+  }
+
+  // Get neuron index by id.
+  ssize_t GetNeuronIndex(ssize_t id) const {
+    for(ssize_t i=0; i < NumNeuron(); ++i) {
+        if (aNeuronGenes[i].Id() == id) {
+            return i;
+        }
+    }
+
+    return -1;  // Id start from 0.
+  }
+
+  // Get link index by innovation id.
+  ssize_t GetLinkIndex(ssize_t innovId) const {
+    for(ssize_t i=0; i < NumLink(); ++i) {
+        if (aLinkGenes[i].InnovationId() == innovId) {
+            return i;
+        }
+    }
+
+    return -1;  // Id start from 0.
+  }
+
+  // Whether link exist and enabled.
+  bool ContainEnabledLink(ssize_t innovId) const {
+    for(ssize_t i=0; i < NumLink(); ++i) {
+        if (aLinkGenes[i].InnovationId() == innovId &&
```

Comment:
If we split the if condition, we could return right after we found the neuron with the specified `innovId`, enabled or not.